src/node.cc

```diff
@@ -2418,6 +2419,7 @@ void DLOpen(const FunctionCallbackInfo<Value>& args) {
     return;
   }
 
+#ifdef ENABLE_NAPI
 // try loading with node 0.10 manner in case of node 0.10
```

Comment:
node_module_old is a structure which reflects the structure of node_module in the 0.10.x release.  We only need it if we are going to try and load a module compiled against 0.10.x which only makes sense with NAPI (without NAPI you have to recompile your module to use it with newer node versions).